app/responders/ropensci/approve_responder.rb

```diff
@@ -0,0 +1,59 @@
+require_relative '../../lib/responder'
+
+module Ropensci
+  class ApproveResponder < Responder
+    keyname :ropensci_approve
+
+    def define_listening
+      @event_action = "issue_comment.created"
+      @event_regex = /\A@#{bot_name} approve( [ \w-]+)?\.?\s*\z/i
+    end
+
+    def process_message(message)
+      return unless verify_package_name
+      return unless verify_submission_type
+
+      update_or_add_value("date-accepted", Time.now.strftime("%Y-%m-%d"), append: false, heading: "Date accepted")
+      respond_external_template(params[:template_file], locals) if params[:template_file]
+      Ropensci::AirtableWorker.perform_async(:clear_assignments, params, locals, { reviewers: list_of_reviewers })
+      Ropensci::ApprovedPackageWorker.perform_async(:new_team, params, locals, { team_name: @package_name })
+      process_labeling
+      close_issue
+    end
+
+    def verify_package_name
+      @package_name = match_data[1].to_s.strip
+      if @package_name.empty?
+        respond("Could not approve. Please, specify the name of the package.")
+        return false
+      end
+      true
+    end
+
+    def verify_submission_type
+      submission_type = read_value_from_body("submission-type").downcase
+      if submission_type == "stats"
+        statsgrade = read_value_from_body("statsgrade").downcase
+        if ["bronze", "silver", "gold"].include?(statsgrade)
+          labels_to_add << "6/approved-#{statsgrade}"
```

Comment:
This bit will need a slight adjustment, which I am unsure how best to implement. The badges are graded, and the labels will be like `6/approved-bronze-v0.0.1`. At the time the command is issued, the version will be the current version which can be obtained from a fairly straightforward URL query. Alternatively, it could easily be incorporated within an actual external service and delivered that way. @xuanxu Let me know if that would be best, and I'll implement once I'm back from vacation after 16th Aug. And then, to complicate matters a bit further, we'll later have some kind of "downgrade" command which will change the grade (like `gold` -> `silver`, while retaining version on current badge and **not** aligning with current version if that differs. But hopefully that can be left to work on once this initial responder is up and running.

Reply:
Maybe instead of adding the label this responder could ping a simple external service passing the grade so the external service can just add the proper label. That would be easy and simple I think.

Reply:
Okay, shall do. Could you please open an issue on [`ropensci-review-tools/roreviewapi`](https://github.com/ropensci-review-tools/roreviewapi) and link back here? Thanks!

Reply:
Done! https://github.com/ropensci-review-tools/roreviewapi/issues/9
I guess this can be merged and the responder updated once the external service is working, I don't see this a stopper to merge this PR, WDYT?

Reply:
Yeah, it'll be fine to merge because the badges with those (non-versioned) labels still don't exist, so command will just fail to add badge, which is fine for the moment.

Reply:
@mpadge @xuanxu should this be merged too, if I followed correctly?

Reply:
@maelle This can be merged, yes. But the labeling with badge grades for the stats-type submissions won't work.